site/README.md

```diff
@@ -0,0 +1,54 @@
+# aip.dev static site generator
+
+Welcome to our NIH static site generator.
+
+**Note:** If you do not care about this and just want to write or modify AIPs,
+see `CONTRIBUTING.md`.
+
+## Why?
+
+GitHub Pages normally automatically builds documentation with [Jekyll][], but
+as the AIP system has grown, we are beginning to reach the limits of what
+Jekyll can handle:
+
+- AIP adoption is handled through fork-and-merge and top-down configuration
+  files will lead to repetitive merge conflicts.
+- Our grouping and listing logic has grown complicated, and had to be
+  maintained using complex and error-prone Liquid templates.
+- Jekyll is extensible but GitHub requires specific Jekyll plugins, meaning we
+  can not use off-the-shelf solutions for planned features (e.g. tabbed code
+  examples).
```

Comment:
But writing our own means that we can't use ANY off the shelf solutions such as:

redirect-from
github-metadata-plugin

It also means we can't use recommended ruby solutions that would be supported by Jekyll. One such Gem I had been looking at using the dotenv Gem and then configuring the gitconfig to ignore the .env file where custom configurations could be stored.

Reply:
However, you can not use those recommended solutions (like the dotenv gem) anyway: you can [only use these dependencies, period](https://pages.github.com/versions/).

I think we may have different views of what "use an off the shelf solution" means. For the tabbed code example, if I want to get that, I have to invent something in Jekyll/Liquid right now. When we move to this world, I can just use a pymarkdown plugin (like [this one](https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/)). I have to add a dependency and make a single line code change and then it works.

It does mean I reimplemented certain glue we had before (like redirects). I view that as a net win because implementing redirects was trivial and implementing these more complex things would be harder.